upgrade.py

```diff
@@ -4,6 +4,8 @@
 import re
 import shutil
 
+from OpenSSL import crypto
```

Comment:
Do we need an additional dependency in order to use `OpenSSL` module?

Reply:
This is right.  Is it too big?  I'm not sure we have many alternatives to choose from.